### DISCUSSION TITLE
Most users cannot see or upload assets (pictures).

### DISCUSSION THREAD

**Discussion by YuriMri** [requarks/wiki#3265] [Error / Bug Report]
**Describe the bug**
Admins made content (LDAP authenticated users in wiki.js Admin group) and other people cannot see uploaded pictures or upload new pictures or other assets.
I tried to create test user (local auth) and gave him all permissions possible (permissions and page rules) for location "path is exactly" or even "page starts with": **/home/company**

Users who are logged in cannot see the picture and see just **picname.png**
After opening console in browser I see:
logo-company.png:1 GET https://wiki-sk.company.eu/logo-company.png 403 (Forbidden)
Image (async)
gi @ app.js?1601761667:350
vi @ app.js?1601761667:350
hi @ app.js?1601761667:350
g @ app.js?1601761667:350
p @ app.js?1601761667:350
f @ app.js?1601761667:350
p @ app.js?1601761667:350
f @ app.js?1601761667:350
p @ app.js?1601761667:350
f @ app.js?1601761667:350
p @ app.js?1601761667:350
f @ app.js?1601761667:350
p @ app.js?1601761667:350
f @ app.js?1601761667:350
p @ app.js?1601761667:350
f @ app.js?1601761667:350
p @ app.js?1601761667:350
f @ app.js?1601761667:350
p @ app.js?1601761667:350
f @ app.js?1601761667:350
p @ app.js?1601761667:350
(anonymous) @ app.js?1601761667:350
t._update @ app.js?1601761667:350
i @ app.js?1601761667:350
fn.get @ app.js?1601761667:350
fn @ app.js?1601761667:350
(anonymous) @ app.js?1601761667:350
Mn.$mount @ app.js?1601761667:350
Mn.$mount @ app.js?1601761667:350
init @ app.js?1601761667:350
(anonymous) @ app.js?1601761667:350
p @ app.js?1601761667:350
f @ app.js?1601761667:350
p @ app.js?1601761667:350
f @ app.js?1601761667:350
p @ app.js?1601761667:350
(anonymous) @ app.js?1601761667:350
t._update @ app.js?1601761667:350
i @ app.js?1601761667:350
fn.get @ app.js?1601761667:350
fn @ app.js?1601761667:350
(anonymous) @ app.js?1601761667:350
Mn.$mount @ app.js?1601761667:350
Mn.$mount @ app.js?1601761667:350
init @ app.js?1601761667:350
(anonymous) @ app.js?1601761667:350
p @ app.js?1601761667:350
(anonymous) @ app.js?1601761667:350
t._update @ app.js?1601761667:350
i @ app.js?1601761667:350
fn.get @ app.js?1601761667:350
fn @ app.js?1601761667:350
(anonymous) @ app.js?1601761667:350
Mn.$mount @ app.js?1601761667:350
Mn.$mount @ app.js?1601761667:350
init @ app.js?1601761667:350
(anonymous) @ app.js?1601761667:350
p @ app.js?1601761667:350
(anonymous) @ app.js?1601761667:350
k @ app.js?1601761667:350
(anonymous) @ app.js?1601761667:350
t._update @ app.js?1601761667:350
i @ app.js?1601761667:350
fn.get @ app.js?1601761667:350
fn.run @ app.js?1601761667:350
un @ app.js?1601761667:350
(anonymous) @ app.js?1601761667:350
Xt @ app.js?1601761667:350
Promise.then (async)
Kt @ app.js?1601761667:350
ie @ app.js?1601761667:350
(anonymous) @ app.js?1601761667:350
fn.update @ app.js?1601761667:350
t.$forceUpdate @ app.js?1601761667:350
p @ app.js?1601761667:350
(anonymous) @ app.js?1601761667:350
(anonymous) @ app.js?1601761667:350
Promise.then (async)
He.p @ app.js?1601761667:350
He @ app.js?1601761667:350
(anonymous) @ app.js?1601761667:350
Be @ app.js?1601761667:350
t._c @ app.js?1601761667:350
eval @ VM313:3
t._render @ app.js?1601761667:350
i @ app.js?1601761667:350
fn.get @ app.js?1601761667:350
fn @ app.js?1601761667:350
(anonymous) @ app.js?1601761667:350
Mn.$mount @ app.js?1601761667:350
Mn.$mount @ app.js?1601761667:350
t._init @ app.js?1601761667:350
Mn @ app.js?1601761667:350
(anonymous) @ app.js?1601761667:19
Show 38 more frames

Also when this user tries to edit page and upload asset (picture) he sees error and I can see in console:
POST https://wiki-sk.company.eu/u 403 (Forbidden)
dt @ editor.js?1601761667:8
(anonymous) @ editor.js?1601761667:8
process @ editor.js?1601761667:8
(anonymous) @ editor.js?1601761667:8
(anonymous) @ editor.js?1601761667:8
Promise.then (async)
(anonymous) @ editor.js?1601761667:8
e @ editor.js?1601761667:8
(anonymous) @ editor.js?1601761667:8
(anonymous) @ editor.js?1601761667:8
c @ editor.js?1601761667:8
(anonymous) @ editor.js?1601761667:8
l @ editor.js?1601761667:8
_write @ editor.js?1601761667:8
(anonymous) @ editor.js?1601761667:8
(anonymous) @ editor.js?1601761667:8
(anonymous) @ editor.js?1601761667:8
e @ editor.js?1601761667:8
requestAnimationFrame (async)
l @ editor.js?1601761667:8
e @ editor.js?1601761667:8
requestAnimationFrame (async)
l @ editor.js?1601761667:8
e @ editor.js?1601761667:8
requestAnimationFrame (async)
l @ editor.js?1601761667:8
e @ editor.js?1601761667:8
requestAnimationFrame (async)
l @ editor.js?1601761667:8
e @ editor.js?1601761667:8
requestAnimationFrame (async)
l @ editor.js?1601761667:8
e @ editor.js?1601761667:8
requestAnimationFrame (async)
l @ editor.js?1601761667:8
e @ editor.js?1601761667:8
requestAnimationFrame (async)
l @ editor.js?1601761667:8
e @ editor.js?1601761667:8
requestAnimationFrame (async)
l @ editor.js?1601761667:8
e @ editor.js?1601761667:8
requestAnimationFrame (async)
l @ editor.js?1601761667:8
e @ editor.js?1601761667:8
requestAnimationFrame (async)
l @ editor.js?1601761667:8
e @ editor.js?1601761667:8
requestAnimationFrame (async)
l @ editor.js?1601761667:8
e @ editor.js?1601761667:8
requestAnimationFrame (async)
l @ editor.js?1601761667:8
e @ editor.js?1601761667:8
requestAnimationFrame (async)
l @ editor.js?1601761667:8
e @ editor.js?1601761667:8
requestAnimationFrame (async)
l @ editor.js?1601761667:8
e @ editor.js?1601761667:8
requestAnimationFrame (async)
l @ editor.js?1601761667:8
e @ editor.js?1601761667:8
requestAnimationFrame (async)
l @ editor.js?1601761667:8
e @ editor.js?1601761667:8
requestAnimationFrame (async)
l @ editor.js?1601761667:8
e @ editor.js?1601761667:8
requestAnimationFrame (async)
l @ editor.js?1601761667:8
e @ editor.js?1601761667:8
requestAnimationFrame (async)
l @ editor.js?1601761667:8
e @ editor.js?1601761667:8
requestAnimationFrame (async)
l @ editor.js?1601761667:8
e @ editor.js?1601761667:8
requestAnimationFrame (async)
l @ editor.js?1601761667:8
e @ editor.js?1601761667:8
requestAnimationFrame (async)
l @ editor.js?1601761667:8
e @ editor.js?1601761667:8
requestAnimationFrame (async)
l @ editor.js?1601761667:8
e @ editor.js?1601761667:8
requestAnimationFrame (async)
l @ editor.js?1601761667:8
e @ editor.js?1601761667:8
requestAnimationFrame (async)
l @ editor.js?1601761667:8
e @ editor.js?1601761667:8
requestAnimationFrame (async)
l @ editor.js?1601761667:8
e @ editor.js?1601761667:8
requestAnimationFrame (async)
l @ editor.js?1601761667:8
e @ editor.js?1601761667:8
requestAnimationFrame (async)
l @ editor.js?1601761667:8
e @ editor.js?1601761667:8
requestAnimationFrame (async)
l @ editor.js?1601761667:8
e @ editor.js?1601761667:8
requestAnimationFrame (async)
l @ editor.js?1601761667:8
e @ editor.js?1601761667:8
requestAnimationFrame (async)
l @ editor.js?1601761667:8
e @ editor.js?1601761667:8
app.js?1601761667:350 {error: {…}, file: {…}}error: body: "Forbidden"code: 403headers: "access-control-allow-credentials: true
↵access-control-allow-origin: https://wiki-sk.company.eu
↵connection: Keep-Alive
↵content-language: en
↵content-length: 86
↵content-type: application/json; charset=utf-8
↵date: Tue, 01 Dec 2020 15:15:55 GMT
↵etag: W/"56-1G2aw8OR/tbhe71SggU04JCvXfM"
↵keep-alive: timeout=5, max=99
↵referrer-policy: same-origin
↵server: Apache/2.4.6 (CentOS) OpenSSL/1.0.2k-fips
↵strict-transport-security: max-age=63072000; includeSubdomains;
↵vary: Origin,Accept-Encoding
↵x-content-type-options: nosniff
↵x-frame-options: deny
↵x-ua-compatible: IE=edge
↵x-xss-protection: 1; mode=block
↵"type: "error"__proto__: Objectfile: abortLoad: ƒ ()abortProcessing: ƒ ()fireSync: ƒ (e)getMetadata: ƒ (e)setMetadata: ƒ (e,t,n)file: (...)fileExtension: (...)fileSize: (...)fileType: (...)filename: (...)filenameWithoutExtension: (...)id: (...)origin: (...)relativePath: (...)serverId: (...)source: (...)status: (...)transferId: (...)get file: ƒ ()get fileExtension: ƒ ()get fileSize: ƒ ()get fileType: ƒ ()get filename: ƒ ()get filenameWithoutExtension: ƒ ()get id: ƒ ()get origin: ƒ ()get relativePath: ƒ ()get serverId: ƒ ()get source: ƒ ()get status: ƒ ()get transferId: ƒ ()__proto__: Object__proto__: Object
Ut @ app.js?1601761667:350
Wt @ app.js?1601761667:350
$t @ app.js?1601761667:350
(anonymous) @ app.js?1601761667:350
Promise.catch (async)
qt @ app.js?1601761667:350
n @ app.js?1601761667:350
qt @ app.js?1601761667:350
t.$emit @ app.js?1601761667:350
click @ app.js?1601761667:350
qt @ app.js?1601761667:350
n @ app.js?1601761667:350
Xi.o._wrapper @ app.js?1601761667:350

**Expected behavior**
see pictures and being able to upload and use assets.

**Host Info (please complete the following information):**
OS:
NAME="CentOS Linux"
VERSION="8 (Core)"
ID="centos"
ID_LIKE="rhel fedora"
VERSION_ID="8"
PLATFORM_ID="platform:el8"
PRETTY_NAME="CentOS Linux 8 (Core)"
ANSI_COLOR="0;31"
CPE_NAME="cpe:/o:centos:centos:8"
HOME_URL="https://www.centos.org/"
BUG_REPORT_URL="https://bugs.centos.org/"
CENTOS_MANTISBT_PROJECT="CentOS-8"
CENTOS_MANTISBT_PROJECT_VERSION="8"
REDHAT_SUPPORT_PRODUCT="centos"
REDHAT_SUPPORT_PRODUCT_VERSION="8"

Wiki.js version: 2.5.159, 
Node.js = 14.6.0
Database engine: postgresql 12.3

**Comment by YuriMri** [comment]
I tested also scenario where I added **testuser** into administrators group and removed him.
For the while he was in admin group the pictures were shown.
As soon as I removed him the pictures disappeared.

**Comment by YuriMri** [comment]
Also tested another thing.
I tried to mess with page rules and found out that this test user can see pictures, but only if rules are set "Page starts with..." and I leave path field blank. 
No matter the rules match if I enter something else. If the field is not blank it will not show pictures for the user.

**Comment by chris-hs** [comment]
Make sure your permissions are correct. The asset permissions are separate from the pages permissions. The user needs read access to assets in order to display them, and write/upload and manage permissions in order to upload and work with assets.

If your assets are stored in the `/assets` folder, give them `read:assets`, `write:assets`, and `manage:assets` permission to that folder. Or if your assets are stored randomly in your wiki, make sure the `read:assets`, `write:assets`, and `manage:assets` apply to the whole wiki.

**Comment by YuriMri** [comment]
Hello,
I now understand that asset permissions are completely separate from pages (asset directory structure).
It was not clear to me from the documentation.
However now I face with problem that there are pictures in root (`/`) and when I set  `read:assets` to "Page starts with... /" everything works, but those users have access to all pictures/assets on wiki.
When I set `read:assets` to "Path is exactly /" users cannot see pictures. As we have some pics in root (company logos and such) I want all users to see but other departments might have pics others should not see.
I might get around this with creating folder "logos" containing such pictures and remapping + setting up permissions.

Now another thing I noticed.
We created some folders for testing assets and we would like to delete them. I did not notice this functionality in GUI and when I rename folder in filesystem (directly in linux on server) the folders will not get renamed. But when I move the file in wiki GUI I see changes in filesystem.
Is there any way to correct this and delete created asset folders so it will not cause instability of the system?
Thank you

**Comment by chris-hs** [comment]
Yeah, if you were going to use the "Path is Exactly" rule, you would have to put the whole path in -- for example `/company_logo.png`, `/picture.jpg`, etc. Using "Page starts with" `/` will match anything in the wiki.

Going by [the documentation](https://docs.requarks.io/guide/structure), the folder structure is created from the file paths. So for example, if you wanted to rename `/folder` to `/folder2`, you would have to rename or move the all the pages in `/folder`. I.e. rename `/folder/page1` to `/folder2/page1`, `/folder/page2` to `/folder2/page2`, etc. However, the **Move / Rename** option is disabled, so that feature must still be in development.

**Comment by YuriMri** [comment]
I was not referring to folders as in URL but rather folders in 'Asset Management'.
![image](https://user-images.githubusercontent.com/69195129/101597032-6ab64400-39f6-11eb-961e-4cf345f375bc.png)
So far I did not find out how can I get rid of those.

**Reply by ghost** [reply]
Apparently asset management is in development.

https://docs.requarks.io/guide/assets
